docker/package/packages.py

```diff
@@ -304,8 +305,11 @@ def mk_node_unit(
     service_file_baker = ServiceFile(
         Unit(after=["network.target"], description="Tezos baker"),
         Service(
+            # The node settings for a generic baker are defined in its own
+            # 'EnvironmentFile', as we can't tell the network from the protocol
+            # alone, nor what node this might connect to
             environment_files=[f"/etc/default/tezos-baker-{proto}"],
-            environment=[f"PROTOCOL={proto}", "NODE_DATA_DIR="],
+            environment=[f"PROTOCOL={proto}"],
```

Comment:
why you keep using `environment` field here instead of `config_file_append`?

Reply:
Good question, I did so for two reasons.

First, the idea of using environment files is that it will make easier for users to change settings without repetition (by editing them). The `PROTOCOL` env var can't be meaningfully modified as it will be the same for all the lifetime of a baker service. 

Secondly, while we could use `config_file_append` we cannot in instantiated baker services, so I didn't do that to be consistent.

The reason we can't in instantiated bakers is that they don't have their own environment file, but instead use the `baking` and `node` ones (this is because `baking` services need these env vars too).
We cannot add `PROTOCOL` to the `node` env file, because that's out of scope and we cannot add it to the `baking` definition because there can be multiple protocol for the same `baking` instance (e.g. on `mainnet` when there is a protocol switch-over there are two).

Reply:
Don't you append variables to configuration file instead of environment file? Why you mention env files here?